src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs

```diff
@@ -124,13 +124,20 @@ private void OnAppRenamed(object sender, RenamedEventArgs e)
             }
             catch (Exception ex)
             {
-                Log.Exception($"OnAppRenamed-{extension}Program|{oldPath}|Unable to create program from {oldPath}", ex, GetType());
+                Log.Exception($"OnAppRenamed-{extension} Program|{e.OldName}|Unable to create program from {oldPath}", ex, GetType());
             }
 
             // To remove the old app which has been renamed and to add the new application.
             if (oldApp != null)
             {
-                Remove(oldApp);
+                if (string.IsNullOrWhiteSpace(oldApp.Name) || string.IsNullOrWhiteSpace(oldApp.ExecutableName) || string.IsNullOrWhiteSpace(oldApp.FullPath))
+                {
+                    Log.Error($"Old app was not initialized properly. OldFullPath: {e.OldFullPath}; OldName: {e.OldName}; FullPath: {e.FullPath}", GetType());
```

Comment:
Same here, an error is something that we have to fix, this seems not such a case.

Reply:
This one should be an error. Because `oldApp` should always have not empty Name, ExecutableName and FullPath. Actually, we get a null reference exception because one of those fields is null.